File: spring/src/main/java/org/sw/sample/exception/CommonException.java
package org.sw.sample.exception;

import org.springframework.http.HttpStatus;

public abstract class CommonException extends RuntimeException {
    private final HttpStatus status;

    public abstract String getReason();

    public CommonException(HttpStatus status, String message) {
        super(message);
        this.status = status;
    }

    public CommonException(HttpStatus status, String message, Throwable cause) {
        super(message, cause);
        this.status = status;
    }

    public HttpStatus getStatus() {
        return status;
    }
}

File: kotlin/src/main/kotlin/org/sw/spring/kotlin/exception/CommonException.kt
package org.sw.spring.kotlin.exception

import org.springframework.http.HttpStatus
import java.lang.RuntimeException

open class CommonException(
    val status: HttpStatus,
    val title: String,
    val instance: String,
    val detail: String,
    cause: Throwable? = null,
) : RuntimeException(
    "$title, $detail", cause
) {
}